pkg/scheduler/api/unschedule_info.go

```diff
@@ -82,12 +82,28 @@ func GetJobOverMaxAllowedMessageForQueue(
 		queueName, resourceNameStr, details)
 }
 
-func GetGangEvictionMessage(taskNamespace, taskName string, minimum int32) string {
+func GetGangEvictionMessage(task *pod_info.PodInfo, job *podgroup_info.PodGroupInfo) string {
+	if len(job.GetSubGroups()) == 1 {
+		if defaultSubgroup, found := job.GetSubGroups()[podgroup_info.DefaultSubGroup]; found {
+			return getGangEvictionForDefaultSubgroupMessage(task.Namespace, task.Name, defaultSubgroup.GetMinAvailable())
+		}
+	}
+	subGroup := job.GetSubGroups()[task.SubGroupName]
+	return getGangEvictionWithSubGroupsMessage(task.Namespace, task.Name, subGroup.GetName(), subGroup.GetMinAvailable())
+}
+
+func getGangEvictionForDefaultSubgroupMessage(taskNamespace, taskName string, minimum int32) string {
 	return fmt.Sprintf(
 		"Workload doesn't have the minimum required number of pods (%d), evicting remaining pod: %s/%s",
 		minimum, taskNamespace, taskName)
 }
 
+func getGangEvictionWithSubGroupsMessage(taskNamespace, taskName, subGroup string, minimum int32) string {
+	return fmt.Sprintf(
+		"Workload doesn't have the minimum required number of pods (%d) on sub-group %s, evicting remaining pod: %s/%s",
```

Comment:
I have a small suggestion here:
"Workload doesn't meet the minimum required number of pods (%d) in sub-group %s. Evicting remaining pod: %s/%s"

If you agree, please also align the message for the default subgroup.